### PR TITLE
Validate config

### PR DESCRIPTION
**Pull request by wagerlabs**

### CODE REVIEW

File: interval.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+			*self = RefreshInterval(time.Duration(i) * time.Second)
+			return nil
+		}
+		d, err := time.ParseDuration(v)
+		if err != nil {
+			return err
+		}
+		ii := RefreshInterval(d)
+		if ii < MinimumRefreshInterval {
+			return fmt.Errorf("Refresh interval must be equal orgreater than or equal to %s", MinimumRefreshInterval)
```
Comment (leonsodhi) [inline interval.go]: Slight typo. This should probably be: "Refresh interval must be greater than or equal to %s"

File: interval.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+		*self = ii
+		return nil
+	default:
+		return fmt.Errorf("Invalid refresh interval: %v", v)
+	}
+}
+
+func (self *RefreshInterval) SetYAML(tag string, v interface{}) bool {
+	err := self.Set(v)
+	return err != nil
```
Comment (leonsodhi) [inline interval.go]: If `self.Set(v)` errors, this should return `false`. I think the opposite is true here. Also, to make it more readable, could we convert this to:

```
if err != nil {		
    return false		
}		
return true
```

File: regex.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+		s, ok := item.(string)
+		if !ok {
+			return false
+		}
+		if err := self.Set(s); err != nil {
+			return false
+		}
+	}
+	return true
+}
```
Comment (leonsodhi) [inline regex.go]: Let's log something if this function errors anywhere.

File: config_manager.go [hunk 4]
```diff
@@ -50,105 +49,28 @@ type ConfigManager struct {
...
-func (cm *ConfigManager) Initialize() error {
-	cm.Config.ExcludeFiles = &RegexCollection{}
-	cm.Config.ExcludePatterns = &RegexCollection{}
 	cm.parseFlags()
-
-	err := cm.readConfig()
-	if err != nil {
-		return err
+	if err := cm.readConfig(); err != nil {
+		log.Errorf("Error reading config file: %v", err)
```
Comment (leonsodhi) [inline config_manager.go]: Do we need to log anything here if we're logging [here](https://github.com/papertrail/remote_syslog2/blob/validate-config/remote_syslog.go#L118)?

File: config_manager.go [hunk 5]
```diff
@@ -50,105 +49,28 @@ type ConfigManager struct {
...
-		os.Exit(1)
-	}
-
-	return cm
-}
-
-func (cm *ConfigManager) Initialize() error {
-	cm.Config.ExcludeFiles = &RegexCollection{}
-	cm.Config.ExcludePatterns = &RegexCollection{}
 	cm.parseFlags()
```
Comment (leonsodhi) [inline config_manager.go]: This should do something with the error returned from `parseFlags()`.

File: config_manager.go [hunk 6]
```diff
@@ -335,21 +262,16 @@ func (cm *ConfigManager) LogLevels() string {
...
-		return cm.Flags.RefreshInterval
-	case cm.Config.RefreshInterval != nil:
-		return *cm.Config.RefreshInterval
-	case cm.Flags.RefreshInterval.Duration != 0:
+	if cm.Flags.RefreshInterval != 0 {
 		return cm.Flags.RefreshInterval
 	}
-	return RefreshInterval{Duration: MinimumRefreshInterval}
+	return MinimumRefreshInterval
 }
```
Comment (leonsodhi) [inline config_manager.go]: This doesn't consider `cm.Config.RefreshInterval`. `cm.Flags.RefreshInterval` should take priority if both are specified.

**Comment by leonsodhi** [comment]
I think the `godeps` dependencies may need updating as I had to pull in `gopkg.in/check.v1` & `/gopkg.in/yaml.v1`.

File: interval.go (hunk 1 above)
Comment (wagerlabs) [inline interval.go]: Fixed

File: interval.go (hunk 2 above)
Comment (wagerlabs) [inline interval.go]: Fixed

File: regex.go (hunk 3 above)
Comment (wagerlabs) [inline regex.go]: Fixed

File: config_manager.go (hunk 4 above)
Comment (wagerlabs) [inline config_manager.go]: Fixed

File: config_manager.go (hunk 5 above)
Comment (wagerlabs) [inline config_manager.go]: Fixed

File: config_manager.go (hunk 6 above)
Comment (wagerlabs) [inline config_manager.go]: Fixed

**Comment by wagerlabs** [comment]
Fixed deps as well.

**Pull request closed without merging**